File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/GeneratorLongPrint.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Control;
import com.systemincloud.modeler.tasks.javatask.api.data.Text;

@JavaTaskInfo(atomic = false)
public class GeneratorLongPrint extends JavaTask {

	@InputPortInfo(name = "Time", dataType = Control.class)
	public InputPort time;
	@InputPortInfo(name = "In", dataType = Control.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Text.class)
	public OutputPort out;

	@Override
	public void executeAsync(InputPort asynchIn) {
		if(asynchIn == time) out.putData(new Text("+"));
		if(asynchIn == in)   out.putData(new Text("-----+"));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/GeneratorLongPrint.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Text import Text

@PythonTaskInfo(atomic=False)
class GeneratorLongPrint(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in")
        self.time = InputPort(name="time")
        self.out = OutputPort(name="out", data_type=Text)

    def execute_async(self, async_in):
        if async_in is self.time:
            self.out.put_data(Text("+"));
        elif async_in is self.in_:
            self.out.put_data(Text("-----+"));

